### ISSUE TITLE
make errorRenderer.draw be optional

### ISSUE THREAD

**Issue by ittayd** [mermaid-js/mermaid#3205]
**Is your feature request related to a problem? Please describe.**
I'm using a mermaid plugin in TiddlyWiki which is an HTML/javascript based wiki. In particular, it allows a live preview of the content. So As I create a mermaid drawing I can see how it looks like and correct mistakes. The problem is that as I type, there are errors of course. The live preview will show them, which is fine, but errorRenderer also adds them at the bottom of the HTML where they accumulate (since it's an HTML wiki the page is not reloaded)

**Describe the solution you'd like**
Making the rendering of errors be optional/configurable

**Comment by ittayd** [comment]
Further, errorRenderer creates an SVG with the same ID as the rendered node and sets a style which now affects the rendered node (in case the calling code uses it. E.g. to put the exception text in it).

**Comment by slorber** [comment]
Same issue here for Docusaurus integration.

We render Mermaid diagrams on a React app and already handle thrown errors with `<ErrorBoundary>` components to display an error message. 

We don't need the weird bomb icon + error message text being appended at the end of the DOM 😅 

Other related issues:
- https://github.com/mermaid-js/mermaid/issues/2317
- https://github.com/mermaid-js/mermaid/issues/4730

--- 

A workaround is to remove the error as soon as it is added to the DOM, cf solution proposed by @sidharthv96 in https://github.com/mermaid-js/mermaid-live-editor/pull/1288

My adapted version:

```js
try {
  return await mermaid.render(id, text);
} catch (e) {
  document.querySelector(`#d${id}`)?.remove();
  throw e;
}
```

Of course it would be much cleaner if there was a way to make the `errorRenderer` optional or configurable here:

https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/mermaidAPI.ts#L478

```js
  // -------------------------------------------------------------------------------
  // Draw the diagram with the renderer
  try {
    await diag.renderer.draw(text, id, version, diag);
  } catch (e) {
    errorRenderer.draw(text, id, version);
    throw e;
  }
```

Is there a good reason to not add `config.errorRenderer` and allow it to be null?

**Comment by sidharthv96** [comment]
https://github.com/mermaid-js/mermaid/pull/4359 is almost there. But some minor issues.

**Comment by silverwind** [comment]
I had hidden those with CSS previously. Ideally I would like to see them off by default, but an option is welcome too.

**Comment by Philipinho** [comment]
For anyone searching, `suppressErrorRendering: true` was released in  [v11.0.0](https://github.com/mermaid-js/mermaid/releases/tag/v11.0.0).

```
mermaid.initialize({
  suppressErrorRendering: true,
});
```
